iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/digitaltwin/DigitalTwinClientTests.java

```diff
@@ -54,6 +61,9 @@
     private DeviceClient deviceClient;
     private DigitalTwinClient digitalTwinClient = null;
     private static final String DEVICE_ID_PREFIX = "DigitalTwinServiceClientTests_";
+    protected static HttpProxyServer proxyServer;
+    protected static String testProxyHostname = "127.0.0.1";
+    protected static int testProxyPort = 8769;
```

Comment:
worth adding a note about our proxy server dependency that requires us to use this particular port

Reply:
I'm not sure I follow what you mean. The port number here is pretty arbitrary

Reply:
I didn't know that the port no here is arbitrary. For .NET, the port no used is dictated by the proxy server configuration used by the test pipeline; which is 8888 for tinyproxy and 3xxx (I don't remember exactly) for squid. Since these are configured to be the listening port for the test proxy server, that is what we need to supply to the client for our e2e tests.
I thought that java followed a similar process, and hence thought that the port no is relevant.

How is the test proxy server configured for java?

Reply:
Oh, in java we spin up the proxy programmatically in the e2e test class itself. In doing so, we can choose an arbitrary port for it to listen on. The variable ```testProxyPort``` here is referenced directly when creating the proxy, and when creating the service client's proxy settings. Because of that, the only real limitation here is that the port has to be a different port from any other e2e test class because we can't open two proxies on one port at the same time, and these tests run in parallel with other test classes

Reply:
For instance: https://github.com/Azure/azure-iot-sdk-java/blob/41b952292a67f88270cd414eb30dd620b48ebd75/iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/setup/SendMessagesCommon.java#L188

Reply:
Got it, that explains! It assume it is a start-the-proxy-server-instance-if-it-isn't-running-already kind of flow?

Reply:
> Got it, that explains! It assume it is a start-the-proxy-server-instance-if-it-isn't-running-already kind of flow?

That's the idea, yeah